test/test_netcdf_output_writer.jl

```diff
@@ -504,10 +504,53 @@ function test_netcdf_function_output(arch)
     return nothing
 end
 
+function test_netcdf_spatial_average(arch)
+    topo = (Periodic, Periodic, Periodic)
+    domain = (x=(0, 1), y=(0, 1), z=(0, 1))
+    grid = RectilinearGrid(arch, topology=topo, size=(4, 4, 4); domain...)
+
+    model = NonhydrostaticModel(grid = grid,
+                                timestepper = :RungeKutta3,
+                                tracers = (:c,), 
+                                coriolis = nothing, 
+                                buoyancy = nothing, 
+                                closure = nothing)
+    set!(model, c=1)
+
+    Δt = 1/64 # Nice floating-point number
+    simulation = Simulation(model, Δt=Δt, stop_time=50Δt)
```

Comment:
Can we do fewe stepsr? Our tests currently stretch the limits of our resources so we need to be as parsimonious as possible when adding new tests. Note also that compilation cost is the main thing. If this test can be combined with another test, that'd be ideal.

Reply:
Also suggest using `stop_iteration`

Reply:
Sorry @glwagner I ended up merging before you commented. I copied the test template for others in the same file so there are more tests that we apply this change to. Would you like me to open another PR for this?

Reply:
Right, I'm pointing out that we don't want to copy/paste test code now without care, since a lot of our test code is poorly written / wasteful and our CI is straining under the pressure... :-/

Reply:
Any PRs that reduce test cost will be greatly appreciated! I can't tell if all the changes will make a big difference, you are better placed to analyze that.

Reply:
After running these a few time for this PR, I'd say that this one won't make much of a difference. But I have identified several tests that instantiate its own model each that could be merged together. That, I think, will have a more significant impact. I'll open a PR about it once some of my other PRs are merged!

Reply:
sounds good